packages/cli-hydrogen/src/cli/services/deploy.ts

```diff
@@ -0,0 +1,97 @@
+import {DeployConfig, ReqDeployConfig} from './deploy/types.js'
+import {createDeployment, healthCheck, uploadDeployment} from './deploy/upload.js'
+import {buildTaskList} from './build.js'
+import {validateProject, fillDeployConfig} from './deploy/config.js'
+import {environment, system, ui} from '@shopify/cli-kit'
+
+interface TaskContext {
+  config: ReqDeployConfig
+  deploymentID: string
+  assetBaseURL: string
+  previewURL: string
+}
+
+const isUnitTest = environment.local.isUnitTest()
+
+export async function deployToOxygen(_config: DeployConfig) {
+  const backoffPolicy = [5, 10, 15, 30, 60]
+
+  await validateProject(_config)
+
+  /* eslint-disable require-atomic-updates */
+  const tasks: ui.ListrTask<TaskContext>[] = [
+    {
+      title: '📝 Getting deployment config',
+      task: async (ctx, task) => {
+        ctx.config = await fillDeployConfig(_config)
+        task.title = '📝 Deployment config parsed'
+      },
+    },
+    {
+      title: '✨ Creating a deployment',
+      task: async (ctx, task) => {
+        const {deploymentID, assetBaseURL} = await createDeployment(ctx.config)
+        ctx.assetBaseURL = assetBaseURL
+        ctx.deploymentID = deploymentID
+        task.title = '✨ Deployment created'
```

Comment:
Yeah, its just a bit tricky to convey this since we have to "create" the deployment on DMS before we upload code to it, but want to provide that feedback to the user incase it fails (or needs to retry, etc)